compiler/rustc_hir_typeck/src/place_op.rs

```diff
@@ -289,13 +289,10 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                         )
                     {
                         let method = self.register_infer_ok_obligations(ok);
-                        let ty::Ref(_, _, mutbl) = *method.sig.output().kind() else {
-                            span_bug!(
-                                self.tcx.def_span(method.def_id),
-                                "expected DerefMut to return a &mut"
-                            );
-                        };
-                        *deref = OverloadedDeref { mutbl, span: deref.span };
+                        if let ty::Ref(_, _, mutbl) = *method.sig.output().kind() {
+                            *deref = OverloadedDeref { mutbl, span: deref.span };
+                        }
```

Comment:
Why was this span_bug removed?

Reply:
It's kinda pointless. I think I mixed up a bit of the refactoring from #132301, but all it's doing is asserting something we've explicitly decided not to care about (which is detecting invalid libcore items, https://github.com/rust-lang/compiler-team/issues/620).

Reply:
But whatever, I'll revert it.